WordPress/gutenberg
# Display settings 'label' defined by the 'register_setting' method

## What?
Resolves #41637.
Alternative #45918.

PR improves UI/UX when editing or saving custom settings defined by third-party integrations.

## Why?
It avoids hardcoding Settings labels map and improves extensibility. See #41637 for more details.

## How?
* Updates Settings API schema to include setting labels. Props to @TimothyBJacobs for suggestions.
* Update the say Site entity config is loaded and added to store. It's similar to post type and taxonomies loaders.
* Settings labels are extracted from the schema and stored in the `entityConfig.meta.labels` property.

## Testing Instructions
1. Register a testing setting.
2. Open a post or page.
3. Insertert a Tagline or Logo block and make changes to it.
4. Change the test setting - `wp.data.dispatch( 'core' ).editEntityRecord( 'root', 'site', undefined, { foo_my_setting: 'Testing' } );`.
7. Multiti entity saving flow should display setting titles, including the custom setting.

### Test setting example
```php
add_action( 'rest_api_init', function() {
	register_setting(
		'general',
		'foo_my_setting',
		array(
			'type'              => 'string',
			'label'             => 'My Setting',
			'description'       => 'This is a custom setting.',
			'show_in_rest'      => true,
			'sanitize_callback' => 'sanitize_text_field',
		)
	);
} );
```

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast
